.github/workflows/e2e-tests.yml

```diff
@@ -1,39 +1,62 @@
 name: E2E Tests
+
 on:
-  push:
-    branches:
-      - main
+    workflow_dispatch:
+    pull_request:
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name for pull requests
+  # or the commit hash for any other events.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
 
 jobs:
-  playwright:
-    name: Playwright Tests
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      - name: Log debug information
-        run: |
-          npm --version
-          node --version
-          git --version
-          php --version
-          composer --version
-
-      - name: Install NodeJS
-        uses: actions/setup-node@v2
-        with:
+    e2e-tests:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            # matrix:
+            #     node: ['14']
+
+        steps:
+        - name: Checkout
+          uses: actions/checkout@v4
+
+        - name: Setup Node
+          uses: actions/setup-node@v4
+          with:
             node-version-file: '.nvmrc'
+            cache: npm
+
+        - name: Install dependencies
+          run: npm ci
+
+        - name: Build plugin
+          run: |
+            npm run build
+
+        - name: Install Playwright dependencies
+          run: |
+                npx playwright install chromium firefox webkit --with-deps
 
-      - name: NPM install
-        run: npm ci --legacy-peer-deps
+        - name: Starting the WordPress Environment
+          run: |
+                npm run wp-env start
 
-      - name: Playwright install
-        run: npx playwright install --with-deps
+        - name: Running the tests
+          env:
+              WP_AUTH_STORAGE: artifacts/storage-states/admin.json
+              WP_BASE_URL: http://localhost:8889
+              WP_USERNAME: admin
+              WP_PASSWORD: password
```

Comment:
Unless you have a hyper custom setup you don't need these, these are all the defaults

Reply:
Removed. Thank you.